docs/modify_fields.md

```diff
@@ -0,0 +1,123 @@
+# `modify_fields` Processor
+
+The `modify_fields` processor allows customization of the structure and contents of log entries.
+
+## Configuration structure
+
+```yaml
+type: modify_fields:
+fields:
+  <destination field>:
+    # Source
+    move_from: <source field>
+    copy_from: <source field>
+    static_value: <string>
+    
+    # Mutation
+    default_value: <string>
+    map_values:
+      <old value>: <new value>
+    type: {integer|float}
+    omit_if: <filter>
+```
+
+The top-level configuration for this processor contains a single field,
+`fields`, which contains a map of output field names and corresponding
+translations.
+
+All transformations are applied in parallel, which means that sources and
+filters operate on the original input log entry and may not reference the new
+value of any other fields being modified by the same processor.
+
+### Source options
+
+#### No source specified
+
+If no source value is specified, the existing value in the destination field
+will be modified.
+
+#### `move_from: <source field>`
+
+The value from `<source field>` will be used as the source for the destination
+field. Additionally, `<source field>` will be removed from the log entry. If a
+source field is referenced by both `move_from` and `copy_from`, the source field
+will still be removed.
+
+#### `copy_from: <source field>`
+
+The value from `<source field>` will be used as the source for the destination
+field. `<source field>` will not be removed from the log entry unless it is also
+referenced by a `move_from` operation or otherwise modified.
+
+#### `static_value: <string>`
+
+The static string `<string>` will be used as the source for the destination field.
+
+### Mutation options
+
+Zero or more mutation operators may be applied to a single field. If multiple
+operators are supplied, they will always be applied in the following order.
+
+#### `default_value: <string>`
+
+If the source field did not exist, the output value will be set to
+`<string>`. If the source field already exists (even if it contains an empty
+string), the original value is unmodified.
```

Comment:
```suggestion
string), the original value is unmodified. This config option is not allowed if the source option is `static_value`.
```

Reply:
I think this is a net negative for readability; it should be obvious that `default_value` is meaningless if the config already specifies a `static_value` from the documentation that's already here. (Since a static value always exists.)

Reply:
It is obvious to me, but I have been involved since the beginning of the design discussion, so I could be biased. We can defer it till Mary reads it, and hopefully it's a fresh view

Reply:
@quentinmit @qingling128 This is confusing. I'm just trying to create a new field with a static value without success. Is that possible with any of this 2 options, and which one? Thank you very much https://stackoverflow.com/questions/73295880/gcp-ops-agent-add-static-field

Reply:
Hi @llermaly , the official documentation of this has been published at https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#logging-processor-modify-fields

In case of a bug, please report it at https://cloud.google.com/support-hub with the following details:
* agent version
* full agent log
* full agent config
* project and VM id
* reproduction steps

Reply:
@llermaly `env` isn't a valid field in a `LogEntry`; you should be able to use `jsonPayload.env` as a field with either `static_value` or `default_value` (the difference is if the input record has `env` set already, `static_value` will overwrite it but `default_value` will not).